Foreign/LibFFI/Base.hs

```diff
@@ -49,6 +49,23 @@ newStructCType cTypes = do
     init_ffi_type ffi_type elements
     return (ffi_type, free ffi_type >> free elements)
 
+sizeAndAlignmentOfCType :: Ptr CType -> IO (Int, Int)
+sizeAndAlignmentOfCType cType = do
+  (size, alignment) <- ffi_type_size_and_alignment cType
+  if size /= 0 && alignment /= 0
+  then return (fromIntegral size, fromIntegral alignment)
+  else do
+    -- The type's size and alignment haven't been initialized
+    -- so we force it with a call to `ffi_get_struct_offsets`.
+    -- When the `offsets` parameter is null, no offsets will be
+    -- returned, instead the type will just be laid out.
```

Comment:
We're no longer using `ffi_get_struct_offsets`, so this comment is now out of date.